pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/ServerTableSizeReader.java

```diff
@@ -80,29 +77,25 @@ public ServerTableSizeReader(Executor executor, HttpConnectionManager connection
           LOGGER.error("Server: {} returned error: {}", instance, getMethod.getStatusCode());
           continue;
         }
-        TableSizeInfo tableSizeInfo = new ObjectMapper().readValue(getMethod.getResponseBodyAsString(), TableSizeInfo.class);
-        serverSegmentSizes.put(instance, tableSizeInfo.segments);
-      } catch (InterruptedException e) {
-        LOGGER.warn("Interrupted exception while reading segment size for table: {}", table, e);
-      } catch (ExecutionException e) {
-        if (Throwables.getRootCause(e) instanceof SocketTimeoutException) {
-          LOGGER.warn("Server request to read table size was timed out for table: {}.", table, e);
-        } else if (Throwables.getRootCause(e) instanceof ConnectTimeoutException) {
-          LOGGER.warn("Server request to read table size timed out waiting for connection. table: {}.", table, e);
-        } else if (Throwables.getRootCause(e) instanceof ConnectionPoolTimeoutException) {
-          LOGGER.warn("Server request to read table size timed out on getting a connection from pool, table: {}.", table, e);
-        } else {
-          LOGGER.warn("Execution exception while reading segment sizes for table: {}.", table, e);
-        }
-      } catch(Exception e) {
-        LOGGER.warn("Error while reading segment sizes for table: {}", table);
+        TableSizeInfo tableSizeInfo = OBJECT_MAPPER.readValue(getMethod.getResponseBodyAsStream(), TableSizeInfo.class);
+        serverToSegmentSizeInfoListMap.put(instance, tableSizeInfo.segments);
+      } catch (Exception e) {
```

Comment:
Was any failure path testing done to ensure this works as expected?

Reply:
Yes, the current test covers error case.